src/data-workspace/get-field-id.js

```diff
@@ -0,0 +1,19 @@
+export const getFieldId = (de, coc) => {
```

Comment:
I decided to do some minor refactoring and made a central place for this. `field.split('.')` and generation of the field-id/name was done in different, and I think it make sense to have this in one place, since we introduced more usage of this in this PR.

Reply:
> What do you think about naming getFieldIdComponents as parseFieldId?

Sure

> getFieldId accepts parameters (deId, cocId) (maybe even as a destructured object e.g. ({ deId, cocId })) and returns a string

Yeah, I thought about this, and also thought about supporting both - eg. checking for `typeof de === 'string'`. However since all usages used/had access to de whole object I just kept it that way to keep it simple. But we can for sure just change it to strings.

> and parseFieldId accepts a string parameter and returns an object { deId, cocId } -- (those can be spelled out or not, depending on what's more consistent with other usage)

Yeah, lets keep it like that, and change the other since;

> and parseFieldId accepts a string and returns the whole metadata objects { dataElement, categoryOptionCombo } (same parenthetical comments as above)

is not straightforward, since you would need access to the metadata.

Reply:
> However since all usages used/had access to de whole object I just kept it that way to keep it simple. But we can for sure just change it to strings.

Let's not implement support for strings until we need to.